lib/navs/thread_main_nav.dart

```diff
@@ -61,17 +70,15 @@ class _MainNavState extends State<ThreadMainNav> {
     );
   }
 
-  void _setSelectedIndex(int selectedIndex) {
-    setState(() {
-      _selectedIndex = selectedIndex;
-    });
+  void _setSelectedIndex(MainNavTab selectedTab) {
+    context.go("/${selectedTab.name}");
   }
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       body: IndexedStack(
-        index: _selectedIndex,
+        index: _getTabIndex(), // post 건너뛰기
```

Comment:
> 탭을 변경해도 스크린의 상태가 유지(OffStage 사용한 것처럼) 되는 이유도 같은 이유일 것 같은데 궁금합니다 .!
 
강의에서처럼 `late final int _selectedIndex = _tabs.indexOf(widget.tab);` 와 같은 방식으로 index 를 넣어주었다면 아마 @0hhanum 님께서도 setState를 해줘야만 탭 변경시마다 화면이 변경 되셨을 테지만 지금은 build 할 때마다 _getTabIndex() 를 실행하기 때문에 문제없이 화면이 변경되고 있습니다.

바꿔 말하면 탭 변경 시마다 build() 가 일어나고 있고 심지어 다른 state 가 변경되어도 해당 함수는 재실행이 된 다는 의미이기 때문에 근원이 되는 widget.currentTab 으로부터 바로 index 값을 가져오는게 더 좋을 것 같습니다.

```suggestion
        index: widget.currentTab.index,
```